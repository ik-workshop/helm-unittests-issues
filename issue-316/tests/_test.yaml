---
tests:
  # - it: should selects the correct kind when autoscaling enabled
  #   set:
  #     autoscaling:
  #       enabled: true
  #   asserts:
  #     - containsDocument:
  #         kind: ScaledObject
  #         apiVersion: keda.sh/v1alpha1

  - it: should not selects the correct kind when autoscaling disabled
    set:
      autoscaling:
        enabled: false
    asserts:
      - containsDocument:
          kind: ScaledObject
          apiVersion: keda.sh/v1alpha1
        not: true

  - it: should have autoscaling enabled with snapshot match
    set:
      autoscaling:
        enabled: true
    asserts:
      - matchSnapshot: {}
