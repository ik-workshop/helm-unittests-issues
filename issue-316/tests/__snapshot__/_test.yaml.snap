should have autoscaling enabled with snapshot match:
  1: |
    apiVersion: keda.sh/v1alpha1
    kind: ScaledObject
    metadata:
      name: foo-scaledobject
    spec:
      pollingInterval: 30
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: foo-deployment
  2: |
    apiVersion: keda.sh/v1alpha1
    kind: TriggerAuthentication
    metadata:
      name: foo-trigger-auth
    spec:
      secretTargetRef:
        key: foo-secret-key
        name: foo-secret
